mkosi/__init__.py

```diff
@@ -1197,6 +1197,7 @@ def luks_setup_all(
         return
     assert loopdev is not None
 
+    root = home = srv = var = tmp = None
```

Comment:
Hmm, this is a fix in the sense that e.g. `root` could be undefined in the `finally` block. But I don't think it's the correct fix, because the code is "off-by-one", the `finally` blocks are attached one level too high. It should be something like this:
```python
root = luks_setup_root(args, loopdev, do_run_build_script)
try:
     …
finally:
     luks_close(root, "Closing LUKS root partition")
```

But I think a better solution would be to define a context manager and do this properly… We should be able to get rid of this crazy nesting then.

Reply:
Turned `luks_open()` into a context manager. Should be a lot cleaner now.